// Copyright (c) 2023 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;
option go_package = "bosdyn/api/synchronized_command";

option java_outer_classname = "SynchronizedCommandProto";

import "bosdyn/api/arm_command.proto";
import "bosdyn/api/gripper_command.proto";
import "bosdyn/api/mobility_command.proto";

message SynchronizedCommand {
    message Request {
        ArmCommand.Request arm_command = 1;
        MobilityCommand.Request mobility_command = 2;
        GripperCommand.Request gripper_command = 3;

    }

    message Feedback {
        ArmCommand.Feedback arm_command_feedback = 1;
        MobilityCommand.Feedback mobility_command_feedback = 2;
        GripperCommand.Feedback gripper_command_feedback = 3;

    }
}
